.github/workflows/conventional-commits.yaml

```diff
@@ -0,0 +1,38 @@
+name: Conventional Commits Validation
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - edited
+
+permissions:
+  contents: read
+  pull-requests: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}  # Ensure that only one instance of this workflow is running per Pull Request
+  cancel-in-progress: true  # Cancel any previous runs of this workflow
+
+jobs:
+  validate-commits:
+    name: Conventional Commits Validation
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dev-build-deploy/commit-me@v1.2.0
+        with:
+          update-labels: false  # OPTIONAL; do not update the Pull Request labels based on the Conventional Commits information.
+          include-commits: true
+          types: |
+            chore
+            ci
+            docs
+            refactor
+            release
+            test
```

Comment:
`fix`, `feat` and `perf` are missing (of which the first two should be included by default; but it might be nice to be explicit).
Since `include-commits: true`, I expect `squash` should also be included here.

Lastly, I believe a [token](https://github.com/dev-build-deploy/commit-me/blob/main/docs/github-action.md#inputs) is needed to validate the commits associated to the PR

Reply:
Makes sense. I did was not certain that it was necessary to make a distinction between `perf` and `fix` and I am on the fence about whether `refactor` and `test` are useful to make distinctions from CI or `fix`. 

Do we gain any additional insights supporting the additional types?

> Since include-commits: true, I expect squash should also be included here.

Fair. I use `squash` all the time and probably do not need it to fail the linter check; though I do want it to be more strict when merging to main

> Lastly, I believe a [token](https://github.com/dev-build-deploy/commit-me/blob/main/docs/github-action.md#inputs) is needed to validate the commits associated to the PR

For some reason I thought it used a default token! Thanks for the pointer.

Reply:
> Do we gain any additional insights supporting the additional types?

I'm not sure; but I do think that since these are conventional, that we should support them.
Especially since [we encourage people to use them](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/CONTRIBUTING.md#use-conventional-commit-messages), CI shouldn't fail on those.

Reply:
> Especially since [we encourage people to use them](https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/CONTRIBUTING.md#use-conventional-commit-messages), CI shouldn't fail on those.

Great catch. I'm comfortable limiting the list to just those included in our CONTRIBUTING documentation. However, I could see `ci` and `release` being very helpful for maintainers, but not for contributors.

Reply:
By default, the Convention Commits specification does not impose any restrictions on the `type` - it only requires `feat` and `fix` as these have a special purpose (in relation to SemVer).

Therefore, `CommitMe` will not impose any restrictions on the `type` unless you provide an additional list of `type`s to restrict to.

Typically, projects are using the Angular definition of types;
- `build`: Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
- `ci`: Changes to our CI configuration files and scripts (examples: CircleCi, SauceLabs)
- `docs`: Documentation only changes
- `feat`: A new feature
- `fix`: A bug fix
- `perf`: A code change that improves performance
- `refactor`: A code change that neither fixes a bug nor adds a feature
- `test`: Adding missing tests or correcting existing tests

(taken from: https://github.com/angular/angular/blob/68a6a07/CONTRIBUTING.md#-commit-message-format)


> [!NOTE]
> The same principle true for the `scopes` configuration parameter (no restrictions, unless configured)